libs/fetcher.js

```diff
@@ -36,12 +35,55 @@ function parseParamValues(params) {
     }, {});
 }
 
+function parseRequest(req) {
```

Comment:
some of these functions could probably move to their own file as well to reduce the fetcher.js size

Reply:
Perhaps with the TS migration we can re-organize the folders. It's clear to me that we have actually to packages here, one for the server, another for the client. It seems wrong to have everything in the same folder as we have right now. So I'm avoiding to split the server code until we don't have things better organized (by the way, `utils` has only client side code now).